### PR TITLE
Add more test cases to increase datalake service test coverage.

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]
Next time please add the previous coverage numbers v/s the one gotten here.

This PR generates coverage report:

Line coverage: | 70.9% (9803 of 13813)
-- | --
11481
28324
40.5% (11481 of 28324)

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=629670&_a=summary&tab=ms.vss-codecoverage-web.code-coverage-details&view=codecoverage-tab

vs previus coverage:

Line coverage: | 69% (9577 of 13874)
-- | --
11246
28362
39.6% (11246 of 28362)

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=628924&view=codecoverage-tab

File: sdk/storage/azure-storage-files-datalake/test/datalake_directory_client_test.cpp [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: MIT
 
 #include "datalake_directory_client_test.hpp"
+#include "azure/storage/common/shared_key_policy.hpp"
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/test/datalake_directory_client_test.cpp:5]: reformat

File: sdk/storage/azure-storage-files-datalake/test/datalake_file_client_test.cpp [hunk 2]
```diff
@@ -2,8 +2,25 @@
 // SPDX-License-Identifier: MIT
 
 #include "datalake_file_client_test.hpp"
+#include "azure/storage/blobs.hpp"
+#include "azure/storage/common/file_io.hpp"
+#include "azure/storage/common/shared_key_policy.hpp"
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/test/datalake_file_client_test.cpp:7]: reformat

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**